src/ringmesh/basic/plugin_manager.cpp

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) 2012-2017, Association Scientifique pour la Geologie et ses
+ * Applications (ASGA). All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of ASGA nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL ASGA BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *     http://www.ring-team.org
+ *
+ *     RING Project
+ *     Ecole Nationale Superieure de Geologie - GeoRessources
+ *     2 Rue du Doyen Marcel Roubault - TSA 70605
+ *     54518 VANDOEUVRE-LES-NANCY
+ *     FRANCE
+ */
+
+#include <ringmesh/basic/plugin_manager.h>
+
+#include <geogram/basic/command_line.h>
+#include <geogram/basic/file_system.h>
+#include <geogram/basic/line_stream.h>
+
+#include <ringmesh/basic/algorithm.h>
+#include <ringmesh/basic/logger.h>
+#include <ringmesh/basic/pimpl_impl.h>
+
+/*!
+ * @file PluginManager class declaration
+ * @author Arnaud Botella
+ */
+
+namespace
+{
+    std::vector< std::string > plugins;
+
+    bool read_plugins_configuration_file( const std::string& configuration_file )
+    {
+        GEO::LineInput file( configuration_file );
+        try
+        {
+            GEO::LineInput file( configuration_file );
+            ringmesh_assert( file.OK() );
+            while( !file.eof() && file.get_line() )
+            {
+                file.get_fields();
+                ringmesh_assert( file.nb_fields() == 1 );
+                RINGMesh::PluginManager::load_module( file.field( 0 ) );
+            }
+        }
+        catch( const std::logic_error& ex )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch a more specific exception instead of a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1181)